kernel/src/process/posix_thread/futex.rs

```diff
@@ -323,11 +347,15 @@ impl FutexItem {
 struct FutexKey {
     addr: Vaddr,
     bitset: FutexBitSet,
+    pid: Pid,
```

Comment:
I think using type `Option<Pid>` here is better. If pid is `None`, then the wait/wake is not private. And adding some docs to describe the behavior here.